functions/system.bash

```diff
@@ -78,49 +70,51 @@ locale_setting() {
     echo "$(timestamp) [openHABian] Setting locale based on user choice... "
     dpkg-reconfigure locales
     loc=$(grep "LANG=" /etc/default/locale | sed 's/LANG=//g')
-    cond_redirect update-locale LANG=$loc LC_ALL=$loc LC_CTYPE=$loc LANGUAGE=$loc
+    cond_redirect update-locale "LANG=$loc LC_ALL=$loc LC_CTYPE=$loc LANGUAGE=$loc"
     whiptail --title "Change Locale" --msgbox "For the locale change to take effect, please reboot your system now." 10 60
     return 0
   fi
 
   echo -n "$(timestamp) [openHABian] Setting locale based on openhabian.conf... "
+  # shellcheck source=/etc/openhabian.conf disable=SC1091
   source "$CONFIGFILE"
   if is_ubuntu; then
-    cond_redirect locale-gen $locales
+    # shellcheck disable=2154
+    cond_redirect locale-gen "$locales"
   else
     touch /etc/locale.gen
     for loc in $locales; do sed -i "/$loc/s/^# //g" /etc/locale.gen; done
     cond_redirect locale-gen
   fi
   cond_redirect dpkg-reconfigure --frontend=noninteractive locales
-  cond_redirect LANG=$system_default_locale; export LANG &>/dev/null
-  cond_redirect LC_ALL=$system_default_locale; export LC_ALL &>/dev/null
-  cond_redirect LC_CTYPE=$system_default_locale; export LC_CTYPE &>/dev/null
-  cond_redirect LANGUAGE=$system_default_locale; export LANGUAGE &>/dev/null
-  cond_redirect update-locale LANG=$system_default_locale LC_ALL=$system_default_locale LC_CTYPE=$system_default_locale LANGUAGE=$system_default_locale
-  if [ $? -eq 0 ]; then echo "OK"; else echo "FAILED"; fi
+  cond_redirect "LANG=${system_default_locale:-en_US.UTF-8}"
+  cond_redirect "LC_ALL=${system_default_locale:-en_US.UTF-8}"
+  cond_redirect "LC_CTYPE=${system_default_locale:-en_US.UTF-8}"
+  cond_redirect "LANGUAGE=${system_default_locale:-en_US.UTF-8}"
+  export LANG LC_ALL LC_CTYPE LANGUAGE
+  if cond_redirect update-locale "LANG=$system_default_locale LC_ALL=$system_default_locale LC_CTYPE=$system_default_locale LANGUAGE=$system_default_locale"; then echo "OK"; else echo "FAILED"; fi
 }
 
 hostname_change() {
   echo -n "$(timestamp) [openHABian] Setting hostname of the base system... "
   if [ -n "$INTERACTIVE" ]; then
-    new_hostname=$(whiptail --title "Change Hostname" --inputbox "Please enter the new system hostname (no special characters, no spaces):" 10 60 3>&1 1>&2 2>&3)
-    if [ $? -ne 0 ]; then return 1; fi
+    if ! new_hostname=$(whiptail --title "Change Hostname" --inputbox "Please enter the new system hostname (no special characters, no spaces):" 10 60 3>&1 1>&2 2>&3); then return 1; fi
```

Comment:
Does not this `if` statement check the "new_hostname" variable? Have you tested this?

Reply:
no didn't test yes I believe you're right will change that back... but did it work before ? the assignment all by itself will be successfull will $? refer to that or to the previous "real" command (whiptail)

Reply:
Just to be clear, I have no experience with `$()` and if statements. We better try this out. I guess we are going to see more of these cases as well.

Reply:
Seems both (old and new) version are fine ... on first test I entered "foobar" twice, on second, I've hit "cancel" two times. 
Therefore will return to my original change.

[11:45:05] openhabian@openhabianpi:~$ cat /tmp/y
#!/bin/bash -x

new_hostname=$(whiptail --title "Change Hostname" --inputbox "Please enter the new system hostname (no special characters, no spaces):" 10 60 3>&1 1>&2 2>&3)
if [ $? -ne 0 ]; then echo "FAIL"; else echo "OK"; fi

if ! new_hostname=$(whiptail --title "Change Hostname" --inputbox "Please enter the new system hostname (no special characters, no spaces):" 10 60 3>&1 1>&2 2>&3); then echo "FAIL"; else echo "OK"; fi
[11:44:48] openhabian@openhabianpi:~$ /tmp/y
++ whiptail --title 'Change Hostname' --inputbox 'Please enter the new system hostname (no special characters, no spaces):' 10 60
+ new_hostname=foobar
+ '[' 0 -ne 0 ']'
+ echo OK
OK
++ whiptail --title 'Change Hostname' --inputbox 'Please enter the new system hostname (no special characters, no spaces):' 10 60
+ new_hostname=foobar
+ echo OK
OK
[11:44:59] openhabian@openhabianpi:~$ /tmp/y
++ whiptail --title 'Change Hostname' --inputbox 'Please enter the new system hostname (no special characters, no spaces):' 10 60
+ new_hostname=
+ '[' 1 -ne 0 ']'
+ echo FAIL
FAIL
++ whiptail --title 'Change Hostname' --inputbox 'Please enter the new system hostname (no special characters, no spaces):' 10 60
+ new_hostname=
+ echo FAIL
FAIL